aws/usageReports/rds/monthlyReport.go

```diff
@@ -0,0 +1,146 @@
+				CpuAverage:           stats.CpuAverage,
+				CpuPeak:              stats.CpuPeak,
+				FreeSpaceMin:         stats.FreeSpaceMin,
+				FreeSpaceMax:         stats.FreeSpaceMax,
+				FreeSpaceAve:         stats.FreeSpaceAve,
+				CostDetail:           detail,
+			}
+		}
+	}
+	return nil
+}
+
+// getRdsMetrics gets credentials, accounts and region to fetch RDS instances stats
+func getRdsMetrics(ctx context.Context, instances []utils.CostPerInstance, aa taws.AwsAccount, startDate, endDate time.Time) (Report, error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	creds, err := taws.GetTemporaryCredentials(aa, RDSStsSessionName)
+	if err != nil {
+		logger.Error("Error when getting temporary credentials", err.Error())
+		return Report{}, err
+	}
+	defaultSession := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(config.AwsRegion),
+	}))
+	account, err := utils.GetAccountId(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when getting account id", err.Error())
+		return Report{}, err
+	}
+	report := Report{
+		Account:    account,
+		ReportDate: startDate,
+		ReportType: "monthly",
+		Instances:  make([]Instance, 0),
+	}
+	regions, err := utils.FetchRegionsList(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when fetching regions list", err.Error())
+		return Report{}, err
+	}
+	instanceChans := make([]<-chan Instance, 0, len(regions))
+	for _, region := range regions {
+		instanceChan := make(chan Instance)
+		go fetchMonthlyInstancesList(ctx, creds, instances, region, instanceChan, startDate, endDate)
+		instanceChans = append(instanceChans, instanceChan)
+	}
+	for instance := range merge(instanceChans...) {
+		report.Instances = append(report.Instances, instance)
+	}
+	return report, nil
+}
+
+// filterRdsInstances filters instances
+func filterRdsInstances(rdsCost []utils.CostPerInstance) ([]utils.CostPerInstance) {
+	costInstances := []utils.CostPerInstance{}
+	for _, instance := range rdsCost {
+		split := strings.Split(instance.Instance, ":")
+		if len(split) == 7 || split[2] != "rds" {
+			costInstances = append(costInstances, utils.CostPerInstance{split[6], instance.Cost})
+		}
+	}
+	return costInstances
+}
+
+// GetRdsMonthlyReport puts a monthly report of RDS in ES
+func GetRdsMonthlyReport(ctx context.Context, rdsCost []utils.CostPerInstance, aa taws.AwsAccount, startDate, endDate time.Time) (error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	logger.Info("Starting RDS utils report", map[string]interface{}{
+		"awsAccountId": aa.Id,
+		"startDate":    startDate.Format("2006-01-02T15:04:05Z"),
+		"endDate":      endDate.Format("2006-01-02T15:04:05Z"),
+	})
+	costInstance := filterRdsInstances(rdsCost)
+	if len(costInstance) == 0 {
+		logger.Info("No RDS instances found in billing data.", nil)
+		return nil
+	}
+	if already, err := utils.CheckAlreadyHistory(ctx, startDate, aa, IndexPrefixRDSReport); already || err != nil {
```

Comment:
Your function `utils.CheckAlreadyHistory` can return an error (if the query execution fails on the ES for instance), and your condition will treat as if a report is already available. You need to check the error separately.